lib/sensors/src/temperature.rs

```diff
@@ -24,13 +27,34 @@ impl<'a, T: HypedI2c> Temperature<'a, T> {
             Ok(_) => Ok(Self {
                 i2c,
                 device_address,
+                calculate_bounds: default_calculate_bounds,
+            }),
+            Err(e) => Err(TemperatureError::I2cError(e)),
+        }
+    }
+
+    /// Create a new instance of the temperature sensor with the specified bounds and attempt to configure it
+    pub fn new_with_bounds(
```

Comment:
what i meant for this is you have `new_with_bounds`, which takes a required bounds function:
```rs
    /// Create a new instance of the temperature sensor with the specified bounds and attempt to configure it
    pub fn new_with_bounds(
        i2c: &'a mut T,
        device_address: TemperatureAddresses,
        calculate_bounds: fn(f32) -> SensorValueRange<f32>,
    ) -> Result<Self, TemperatureError> {
        // Set up the temperature sensor by sending the configuration settings to the STTS22H_CTRL register
        let device_address = device_address as u8;
        match i2c.write_byte_to_register(device_address, STTS22H_CTRL, STTS22H_CONFIG_SETTINGS) {
            Ok(_) => Ok(Self {
                i2c,
                device_address,
                calculate_bounds,
            }),
            Err(e) => Err(TemperatureError::I2cError(e)),
        }
    }
```
and then `new` just needs to be:
```rs
    /// Create a new instance of the temperature sensor and attempt to configure it
    pub fn new(
        i2c: &'a mut T,
        device_address: TemperatureAddresses,
    ) -> Result<Self, TemperatureError> {
        Self::new_with_bounds(i2c, device_address, default_calculate_bounds)
    }
```
which means no code duplication and no optional args anywhere

Reply:
ah crap, optional still being there was just silly copy+paste mistake lmao